routing_model/src/scenario_tests.rs

```diff
@@ -1005,6 +1020,16 @@ mod src_tests {
                         candidate: CANDIDATE_130,
                     }),
                 },
+                check_and_process_elder_change_routine: CheckAndProcessElderChangeState {
+                    sub_routine_process_elder_change: Some(ProcessElderChangeState {
+                        change_elder: ChangeElder {
+                            changes: vec![(YOUNG_ADULT_205, true), (NODE_ELDER_130, false)],
+                            new_section: SECTION_INFO_1,
+                        },
+                        wait_votes: Vec::new(),
+                    }),
+                    ..Default::default()
+                },
```

Comment:
@jeanphilippeD I added these blocks, they seem correct no? But I'm not 100% sure

Reply:
That does not seem right, if the state was not there before not clear why it is here now

Reply:
I think the difference is from 
old code:
`let change_elder = state.mut_routine_state().change_elder.take().unwrap();`
new code:
`let change_elder = state.mut_routine_state().change_elder.clone();`

Reply:
```Some(state.mark_elder_change().start_check_elder_timeout())```
This is your problem, we do not exit the sub routine there, but we should have.
```wait_votes: Vec::new()``` means that we completed the tasks, so that sub routine should not exist anymore.

Reply:
yep I'm adding the exit_event_loop logic now